queue.go

```diff
@@ -165,6 +166,7 @@ func (q *Queue) Put(val interface{}, timeout float64) error {
 	q.clearPending()
 	isfull := q.isfull()
 	if timeout < 0.0 && isfull {
+		q.mutex.Unlock()
```

Comment:
I just want to simulate the scene where I found the bug yet the test case I writed is poor . you fix it and I will close the pr.